pkg/rt/manager.go

```diff
@@ -102,47 +124,59 @@ func (m *Manager) Refresh() {
 
 // Active returns the set of service IDs currently being managed.
 // Mostly useful for tests.
-func (m *Manager) Active() (serviceIDs []string) {
+func (m *Manager) Active() []string {
+	serviceIDs := []string{}
 	m.mtx.RLock()
 	defer m.mtx.RUnlock()
-	return m.managedIDsWithLock()
+	for _, key := range m.managedKeysWithLock() {
+		serviceIDs = append(serviceIDs, key.serviceID)
+	}
+	return serviceIDs
 }
 
 // StopAll terminates and cleans up all active subscribers.
 func (m *Manager) StopAll() {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
 
-	for _, id := range m.managedIDsWithLock() {
-		level.Info(m.logger).Log("service_id", id, "subscriber", "stop")
-		irq := m.managed[id]
+	for _, key := range m.managedKeysWithLock() {
+		level.Info(m.logger).Log("service_id", key.serviceID, "type", key.product, "subscriber", "stop")
+		irq := m.managed[key]
 		irq.cancel()
 		for i := 0; i < cap(irq.done); i++ {
 			err := <-irq.done
-			level.Debug(m.logger).Log("service_id", id, "goroutine", i+1, "of", cap(irq.done), "interrupt", err)
+			level.Debug(m.logger).Log("service_id", key.serviceID, "goroutine", i+1, "of", cap(irq.done), "interrupt", err)
 		}
-		delete(m.managed, id)
+		delete(m.managed, key)
 	}
 }
 
-func (m *Manager) spawn(serviceID string) interrupt {
+func (m *Manager) spawn(serviceID string, product string) interrupt {
 	var (
 		subscriber  = NewSubscriber(m.client, m.token, serviceID, m.metrics.MetricsFor(serviceID), m.subscriberOptions...)
 		ctx, cancel = context.WithCancel(context.Background())
-		done        = make(chan error, 2)
+		done        = make(chan error, 1)
 	)
-	go func() { done <- fmt.Errorf("realtime: %w", subscriber.RunRealtime(ctx)) }()
-	go func() { done <- fmt.Errorf("origins: %w", subscriber.RunOrigins(ctx)) }()
+	switch product {
+	case api.OriginInspector:
+		go func() { done <- fmt.Errorf("origins: %w", subscriber.RunOrigins(ctx)) }()
+	default:
+		go func() { done <- fmt.Errorf("realtime: %w", subscriber.RunRealtime(ctx)) }()
+	}
+
 	return interrupt{cancel, done}
 }
 
-func (m *Manager) managedIDsWithLock() []string {
-	ids := make([]string, 0, len(m.managed))
-	for id := range m.managed {
-		ids = append(ids, id)
+func (m *Manager) managedKeysWithLock() []subscriberKey {
```

Comment:
I wonder if we need this function all. The only users are in this file and none of them need a sorted slice for anything.